packages/valory/customs/uniswap_pools_search/uniswap_pools_search.py

```diff
@@ -417,6 +419,255 @@ def calculate_metrics(current_pool: Dict[str, Any], coingecko_api_key: str, coin
         "sharpe_ratio": sharpe_ratio
     }
 
+
+# # # # # New Liquidity Analytics functions 
+
+# API details - these value required to  run the functions 
+EY = ""  # Replace with your API key
+SUBGRAPH_URL = f"https://gateway.thegraph.com/api/{EY}/subgraphs/id/5zvR82QoaXYFyDEKLZ9t6v9adgnptxYpKpSbxtgVENFV"
+# 
+# Specific pool ID to analyze
+POOL_ID = "0x8ad599c3a0ff1de082011efddc58f1908eb6e6d8"
+# 
+# Constants
+PRICE_IMPACT = 0.01  # 1% standard price impact
+MAX_POSITION_BASE = 50  # Base for maximum position calculation
+
+
+def fetch_pool_data(pool_id: str) -> Optional[Dict[str, Any]]:
+    """
+    Fetch pool data for a specific pool ID from The Graph.
+
+    Args:
+        pool_id (str): The unique identifier of the pool to fetch data for.
+
+    Returns:
+        Optional[Dict[str, Any]]: A dictionary containing pool data, or None if retrieval fails.
+    """
+    query = {
+        "query": f"""
+        {{
+          pool(id: "{pool_id.lower()}") {{
+            id
+            token0 {{
+              id
+              symbol
+              decimals
+            }}
+            token1 {{
+              id
+              symbol
+              decimals
+            }}
+            liquidity
+            totalValueLockedUSD
+            totalValueLockedToken0
+            totalValueLockedToken1
+          }}
+        }}
+        """
+    }
+    try:
+        logging.info(f"Fetching data for pool ID: {pool_id}")
+        response = requests.post(
+            SUBGRAPH_URL,
```

Comment:
we already have a dictionary of graphql_endpoints mapped with chain name, use that instead